### DISCUSSION TITLE
V2 auth 1.0a: Forbidden 403 when creating tweet

### DISCUSSION THREAD

**Discussion by MohammadA98** [tweepy/tweepy#1797] [Questions] [answered]
Keep running to 403 error when running, cannot seem to find an error in my code as I followed the docs

Generated keys multiple times, changed access to elevated, changed permission to read and write but not sure what to put for callback URL so i used the regular twitter homepage link. 
```
import tweepy

client = tweepy.Client(
    consumer_key=consumer_key, consumer_secret=consumer_secret,
    access_token=access_token, access_token_secret=access_token_secret
)

client.create_tweet(
    text="Ahem Ahem, Testing!"
)
print(f"https://twitter.com/user/status/{response.data['id']}")
```

**Comment by Harmon758** [comment]
What's the full traceback?
Have you regenerated your credentials after giving your app the write permission?

**Comment by MohammadA98** [comment]
> What's the full traceback? Have you regenerated your credentials after giving your app the write permission?

The traceback is as follows: 

```
Traceback (most recent call last):
  File "/Users/mohammad/Desktop/Dislike Bot/authentice.py", line 42, in <module>
    client.create_tweet(
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/tweepy/client.py", line 563, in create_tweet
    return self._make_request(
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/tweepy/client.py", line 168, in _make_request
    response = self.request(method, route, params=request_params,
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/tweepy/client.py", line 124, in request
    raise Forbidden(response)
tweepy.errors.Forbidden: 403 Forbidden
```

I would regenerate my credentials but I need to understand what to include for the Callback URI / Redirect URL.

**Comment by Harmon758** [comment, marked as answer]
It shouldn't matter if you're simply authenticating as your own developer account. The Callback URI / Redirect URL is for 3-legged OAuth. You said you already gave your app the write permission, so you just need to regenerate and use new credentials to utilize it. See the [relevant FAQ section in the documentation](https://tweepy.readthedocs.io/en/v4.5.0/faq.html#why-am-i-encountering-a-401-unauthorized-error-with-api-or-403-forbidden-error-with-client) about this.

**Comment by MohammadA98** [comment]
Alright thank you that worked!
